docs/Adding-Software-to-Homebrew.md

```diff
@@ -8,46 +8,139 @@ Before you start, please check the open pull requests for [Homebrew/homebrew-cor
 
 ## Formulae
 
+**Note:** Before taking the time to craft a new formula:
+
+* make sure it can be accepted by checking [Acceptable Formulae](Acceptable-Formulae.md).
+* check that the formula was not [already refused](https://github.com/Homebrew/homebrew-core/search?q=is%3Aclosed&type=Issues).
+* if you are just updating the version of an existing formula, see the streamlined method in ['submit a new version of an existing formula'](How-To-Open-a-Homebrew-Pull-Request.md#submit-a-new-version-of-an-existing-formula).
+
 ### Writing the formula
 
-1. It's a good idea to find existing formulae in Homebrew that have similarities to the software you want to add. This will help you to understand how specific languages, build methods, etc. are typically handled. Start by tapping `homebrew/core`: first set `HOMEBREW_NO_INSTALL_FROM_API=1` in your shell environment, then run `brew tap homebrew/core` to clone the `homebrew/core` tap to the path returned by `brew --repository homebrew/core`.
+Making a new formula is easy, and the [Formula Cookbook](Formula-Cookbook.md) is an essential resource to guide you through the process. The Cookbook provides detailed instructions, best practices, and solutions to common problems you may encounter when creating a formula. Familiarizing yourself with its contents before you start is advisable.
 
-1. If you're starting from scratch, you can use the [`brew create` command](Manpage.md#create-options-url) to produce a basic version of your formula. This command accepts a number of options and you may be able to save yourself some work by using an appropriate template option like `--python`.
+1. Begin by researching existing formulae in Homebrew that are similar to the software you plan to add. This will help you grasp how specific languages, build methods, and other typical conventions are handled in formulae. Start by setting `HOMEBREW_NO_INSTALL_FROM_API=1` in your shell environment. Then, run `brew tap homebrew/core` to clone the `homebrew/core` tap to the directory indicated by `brew --repository homebrew/core`.
 
-1. You will now have to develop the boilerplate code from `brew create` into a full-fledged formula. Your main references will be the [Formula Cookbook](Formula-Cookbook.md), similar formulae in Homebrew, and the upstream documentation for your chosen software. Be sure to also take note of the Homebrew documentation for writing [Python](Python-for-Formula-Authors.md) and [Node](Node-for-Formula-Authors.md) formulae, if applicable.
+1. If you're starting from scratch, you can use the [`brew create` command](Manpage.md#create-options-url) to produce a basic version of your formula. This command accepts a number of options and you may be able to save yourself some work by using an appropriate template option like `--python` / `--go`.
```

Comment:
```suggestion
1. If you're starting from scratch, you can use the [`brew create` command](Manpage.md#create-options-url) to produce a basic version of your formula. This command accepts a number of options and you may be able to save yourself some work by using an appropriate template option like `--python` or `--go`.
```

Reply:
Fixed in 1de9399aa7fb529f1387491ba1359ea71be5f831